samcli/commands/delete/delete_context.py

```diff
@@ -219,19 +232,23 @@ def delete_ecr_companion_stack(self):
 
             retain_repos = self.ecr_repos_prompts(ecr_companion_stack_template)
 
-            # Delete the repos created by ECR companion stack if not retained
-            ecr_companion_stack_template.delete(retain_resources=retain_repos)
-
             click.echo(f"\t- Deleting ECR Companion Stack {self.companion_stack_name}")
             try:
                 # If delete_stack fails and its status changes to DELETE_FAILED, retain
                 # the user input repositories and delete the stack.
                 self.cf_utils.delete_stack(stack_name=self.companion_stack_name)
                 self.cf_utils.wait_for_delete(stack_name=self.companion_stack_name)
                 LOG.debug("Deleted ECR Companion Stack: %s", self.companion_stack_name)
+
+                # Delete the repos created by ECR companion stack if not retained
+                ecr_companion_stack_template.delete(retain_resources=retain_repos)
```

Comment:
shall we try to delete the companion stack ecr repositories first before trying to delete the stack

Reply:
If termination protection is enabled for the ECR companion stack, we don't want to delete the ECR repositories.